### DISCUSSION TITLE
setup method of pytest is not supported

### DISCUSSION THREAD

**Discussion by zh199609** [seleniumbase/SeleniumBase#2645] [General]
```python
class FailingTests(BaseCase):

    def setup(self):
        print('this is setup')
``` 
run As：
```python
if __name__ == '__main__':
    pytest.main(['test_fail.py', '-v'])
``` 

Why does pytest's setup fail to run

**Comment by zh199609** [comment]
```python
    def setup_class(self):
        print('this is setup_class')
``` 

When the setup_class method of pytest is written, selenium base cannot work anymore

**Comment by mdmintz** [comment]
See [examples/boilerplates/base_test_case.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/boilerplates/base_test_case.py) for an example of customizing  `setUp()` and `tearDown()` properly.

Also see https://github.com/seleniumbase/SeleniumBase/discussions/2625
